packages/mdl-ripple/index.js

```diff
@@ -39,67 +21,123 @@ export default class MDLRipple {
-      [Identifier.SURFACE]: surface,
-    };
-
     for (let i = 0; i < MAX_RIPPLES; i++) {
       let foreground = document.createElement('div');
       foreground.classList.add(Class.FOREGROUND);
       let foregroundCircle = document.createElement('div');
       foregroundCircle.classList.add(Class.FOREGROUND_CIRCLE);
       foreground.appendChild(foregroundCircle);
       root.appendChild(foreground);
+    }
+    return root;
+  }
 
-      this.elements_[this.get(Identifier.FOREGROUND, i)] = foreground;
-      this.elements_[this.get(Identifier.FOREGROUND_CIRCLE, i)] = foregroundCircle;
+
+  /**
+   * @param {!Element} surface
+   * @param {!Element} root
+   * @return {!MDLRipple}
+   */
+  static attachTo(surface, root) {
+    let elements = {
+      [Identifier.SURFACE]: surface,
+      [Identifier.ROOT]: root,
+      [Identifier.BACKGROUND]: root.getElementsByClassName(Class.BACKGROUND)[0]
+    };
+
+    let foregrounds = root.getElementsByClassName(Class.FOREGROUND);
+    let foregroundCircles = root.getElementsByClassName(Class.FOREGROUND_CIRCLE);
+    for (let i = 0; i < foregrounds.length; i++) {
+      elements[ref(Identifier.FOREGROUND, i)] = foregrounds[i];
+      elements[ref(Identifier.FOREGROUND_CIRCLE, i)] = foregroundCircles[i];
     }
+
+    return new MDLRipple(elements);
   }
 
+
+  /** @param {!Object} elements */
+  constructor(elements) {
+    super(elements[Identifier.ROOT]);
+
+    this.elements_ = elements;
+
+    this.initMdlRipple_();
+
+    let surface = elements[Identifier.SURFACE];
+
+    this.isBounded = surface.getAttribute('bounded') != 'false';
+    if (surface.hasAttribute('max-radius')) {
+      this.maxRadius = parseInt(surface.getAttribute('max-radius'), 10);
+    }
+
+    surface.addEventListener('mousedown', (ev) => {
+      let coords = getNormalizedEventCoords(ev, surface);
+
+      this.startTouchBeganAnimationAtPoint(coords.top, coords.left);
+    });
+    surface.addEventListener('mouseup', (ev) => {
+      let coords = getNormalizedEventCoords(ev, surface);
+
+      this.startTouchEndedAnimationAtPoint(coords.top, coords.left);
+    });
+    surface.addEventListener('mouseout', () => {
+      this.cancelAnimations();
+    });
+  }
+}
+
+
+MDLRippleMixin.call(MDLRipple.prototype, {
   getRootElement() {
     return this.elements_[this.get(Identifier.ROOT)];
-  }
+  },
 
   getBoundingClientRect(id) {
     return this.elements_[id].getBoundingClientRect();
-  }
+  },
 
   setStyles(id, styles) {
     Object.assign(this.elements_[id].style, styles);
-  }
+  },
 
   forceLayout(id) {
     this.elements_[id].offsetWidth;
-  }
+  },
 
   addClass(id, cls) {
     this.elements_[id].classList.add(cls);
-  }
+  },
 
   removeClass(id, cls) {
     this.elements_[id].classList.remove(cls);
-  }
+  },
 
-  /**
-   * Element refs are concatenated symbol paths of arbitrary length.
-   * e.g.
-   * get('md-ripple__foreground', 0) = `md-ripple__foreground|0`
-   */
   get(...symbolPath) {
```

Comment:
do you think this would be a bit confusing for authors to implement? We'd probably have to define pretty clearly what constitutes a symbol path and how frameworks could figure out how to locate an element given its symbol path. 

Would a `querySelector` wrapper be easier? e.g. `mdl-ripple__foreground:first-child`